plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/controller/ChatController.kt

```diff
@@ -536,5 +572,7 @@ class ChatController private constructor(
             .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
             .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
             .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+
+        private var projectContextServiceInit = false
```

Comment:
this is global to application

Reply:
OK. What is unique to a workspace? I can put that to make it run once per workspace.

Reply:
think we should put the entire logic into a project level service

Reply:
Which service? Can you help me locate that?

Reply:
https://github.com/aws/aws-toolkit-jetbrains/pull/4858/files#r1752719530

Reply:
let me know if it works

Reply:
It works. Thanks!